### PR TITLE
[mlnx-fw-upgrade] Add FW reactivation in case 2 FW upgrades were done without reboot

### PR DESCRIPTION
**Pull request by volodymyrsamotiy**
#### Why I did it
In order to activate FW after it was upgraded need to perform reboot.
If reboot wasn't performed and user need  to upgrade to another SONiC image then it will fail.
The reason for that is that during SONiC upgrade new FW should be installed but it will fail because previously installed FW wasn't activated.
In order to allow 2nd FW upgrade without reboot in-between need to reactivate FW image.
This change handles such flow.

Example of issue scenario:
- User installed SONiC image on the switch
- Then for some reason FW was upgraded by user or script but reboot was not performed to activate it.
- After that upgrade to new SONiC image will fail because new image need to install FW but it fails due to previous one wasn't activated.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
In "mlnx-fw-upgrade" script check if FW upgrade failed with the error that FW was already installed but reboot was not performed.
If so then perform FW image reactivation and try to upgrade FW again.

#### How to verify it

- Install SONiC image on the switch
- Then upgrade FW but don't perform reboot.
- After that upgrade to new SONiC image and check that upgrade was successfull.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [x] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ] 

#### Description for the changelog

[mlnx-fw-upgrade] Add FW reactivation in case 2 FW upgrades were done without reboot

### CODE REVIEW

**Comment by dgsudharsan** [comment]
@liushilongbuaa The ms_conflict fails again. Can you please resolve

**Review by liat-grozovik — APPROVED** [review]

**Review by stepanblyschak — APPROVED** [review]

File: platform/mellanox/mlnx-fw-upgrade.j2 [hunk 1]
```diff
@@ -243,6 +244,22 @@ function RunFwUpdateCmd() {
     fi
 
     ERROR_CODE="$?"
+
+    if [[ "${ERROR_CODE}" == "${FW_ALREADY_UPDATED_FAILURE}" ]]; then
```
Comment (vdahiya12) [inline platform/mellanox/mlnx-fw-upgrade.j2:248]: Can you please explain the newly defined error code is matched here?

**Review by vdahiya12 — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@StormLiangMS  Tested on top of 202305 commit https://github.com/sonic-net/sonic-buildimage/commit/5c787191ff346c694033547f643f81969027e05b

**Pull request merged**